.github/workflows/docker-build-and-push.yml

```diff
@@ -0,0 +1,59 @@
+# Supply DOCKER_DEPLOY_TAG environment variable from calling workflow (e.g., `staging` or `production`)
+name: Docker Build and Push
+
+on:
+  workflow_call:
+
+jobs:
+  docker-build-push:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_REGISTRY: docker.cdot.systems
+      DOCKER_USERNAME: telescope-ci
+    strategy:
+      matrix:
+        include:
+          # TODO
+          # - context: src/api/Dockerfile
+          #  image: telescope-base
```

Comment:
This one gonna be un-commented once we change all our services to use the base Image right?

Reply:
I'm not 100% sure how to deal with the build order for this one, so I left it for now to do in a follow-up.